### ISSUE TITLE
When creating XFS root partion with calamares, it's not using big timestamps option

### ISSUE THREAD

**Issue by shmerl** [calamares/calamares#1874]
I tried installing Debian testing using Calamares installer, and noticed that resulting root XFS partition is not configured with `bigtime=1` which results in a bunch of warnings in dmesg that usage is limited until 2038. I suppose `mkfs.xfs` really should switch to that by default, but until that happens, it would be good to explicitly use that when creating partitions.

I.e. something like `mkfs.xfs -m bigtime=1 ...`

**Comment by demmm** [comment]
AFAIK, this is an issue for kpmcore, mkfs.xfs comes from there.
https://invent.kde.org/search?search=mkfs.xfs&group_id=1561&project_id=18&scope=&search_code=true&snippets=false&repository_ref=master&nav_source=navbar
https://github.com/calamares/calamares/search?q=xfs
More info regarding bigtime:
https://www.phoronix.com/scan.php?page=news_item&px=XFS-Linux-5.10
@stikonas ^

**Comment by adriaandegroot** [comment]
We use the kpmcore operation for creating the filesystem:

```
CreateFileSystemOperation( *m_device, *m_partition, m_partition->fileSystem().type())
```

There does not seem to be a way to set options (in kpmcore) on the creation of a filesystem. As a **workaround** we could run `xfs_admin -O bigtime=1` after creating the filesystem.

**Comment by demmm** [comment]
>There does not seem to be a way to set options (in kpmcore) on the creation of a filesystem.

That is why I asked for @stikonas opinion to see if it can be added as default in kpmcore

**Comment by adriaandegroot** [comment]
.. note that on the Debian 11 live ISO I have, there are no xfsprogs installed by default. After installing them with `apt install xfsprogs`, I end up with version 5.10.0-4. This version does not support `xfs_admin -O`, and the `xfs_repair` executable does not recognize `-c bigtime=1` , even if the `mkfs.xfs` does recognize metadata flag `-m bigtime=1`.

(For reference, https://git.kernel.org/pub/scm/fs/xfs/xfsprogs-dev.git/tree/db/xfs_admin.sh is the shell script, notice that `-O` is a recent 2021 addition, and calls out to `xfs_repair` .. in other words, for **Debian** the workaround isn't even going to work because the available tooling is too old, but we can try to apply it)

**Comment by shmerl** [comment]
Thanks, Debian should catch up with xfsprogs, at least if there is a live image for Debian testing, that should already take care of it.

Regarding running `xfsadmin`, the man page says one has to make sure the filesystem is in consistent state before doing that. I guess it's not a high chance it will be corrupted right after creation, but normally you'd run a check and possibly repair first:

> NOTE: Administrators must ensure the filesystem is clean by running xfs_repair -n to inspect the filesystem before performing the upgrade.  If corruption is found, recovery procedures (e.g.  reformat  followed  by  restoration  from            backup; or running xfs_repair without the -n) must be followed to clean the filesystem.

**Comment by stikonas** [comment]
In principle it is possible to enable `bigtime=1` by default. But then those XFS partitions would not be mountable on Linux 5.9 and older which is just over one year old. I'm not using XFS myself, so it's a bit hard for me to tell what's the normal usage pattern for XFS is. Is it usually used for rootfs or do people sometimes use it for removable devices too?

**Comment by shmerl** [comment]
XFS is root is probably the most common use case. Removable devices are more commonly something like exfat to make them readable as widely as possible.

**Comment by stikonas** [comment]
Looks like it's being enabled by default in XFS:
https://git.kernel.org/pub/scm/linux/kernel/git/djwong/xfsprogs-dev.git/commit/?h=xfsprogs-5.15-fixes&id=49dc99b3af6914b09cd31da6bd31eb9e307e325a
